tests/test_particle_sets.py

```diff
@@ -291,3 +291,23 @@ def SetLat(particle, fieldset, time, dt):
     assert np.allclose([p.lon for p in pset], lon)
     assert np.allclose([p.lat for p in pset], .6)
     assert np.allclose([p.time for p in pset], 0)
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+@pytest.mark.parametrize(('npart','interval'),[(1,0.2),(5,0.2),(1,1.0),(5,1.0)])
+def test_variable_written_once(fieldset, mode, interval, tmpdir, npart):
+    filepath = tmpdir.join("pfile_once_written_variables")
+    def Update_v(particle, fieldset, time, dt):
+        particle.v_once += 1.
+    def SetLat(particle, fieldset, time, dt):
+        particle.lat = .6
+    class MyParticle(ptype[mode]): 
+        v_once = Variable('v_once', dtype=np.float32, initial=1.,to_write='once')
+    lon = np.linspace(0, 1, npart, dtype=np.float32)
+    lat = np.linspace(1, 0, npart, dtype=np.float32)
+    pset = ParticleSet(fieldset, pclass=MyParticle, lon=lon, lat=lat)
+    pset.execute(pset.Kernel(SetLat)+pset.Kernel(Update_v), starttime=0., endtime=1, dt=0.1, interval=interval,output_file=pset.ParticleFile(name=filepath))
+    ncfile = Dataset(filepath+".nc", 'r', 'NETCDF4') 
+    V_once = ncfile.variables['v_once'][:]
+    assert (V_once.shape == (npart,))
+    assert (V_once[0] == 1.)
```

Comment:
Also assert that `pset[0].v_once == 11`, just to check that the actual particle attribute has been updated?